src/IceRpc/Transports/Tcp/Internal/TcpListener.cs

```diff
@@ -17,6 +17,9 @@ internal sealed class TcpListener : IListener<IDuplexConnection>
     private readonly MemoryPool<byte> _pool;
     private readonly Socket _socket;
 
+    // Set to 1 when the listener is disposed.
+    private int _disposed;
```

Comment:
I don't understand how this "int _disposed" flag works. Should it be volatile? Why not use a bool?

Reply:
> I don't understand how this "int _disposed" flag works. Should it be volatile? 

Yes Using volatile makes sense. Here the flag allow us to see if `SocketError.OperationAborted` is throw because the listener was disposed.

> Why not use a bool?

There is no bool overload for `Interlocked.Exchange` and value types are not supported by the generic `Interlocked.Exchange` in .NET 8, this restriction was removed in .NET 9

https://github.com/dotnet/runtime/commit/41e02e5549a7954b068f342cd9bd6611ef764cc3